src/Compilers/Core/Portable/Operations/OperationExtensions.cs

```diff
@@ -10,6 +11,22 @@ namespace Microsoft.CodeAnalysis.Semantics
 {
     public static class OperationExtensions
     {
+        /// <summary>
+        /// This will check whether context around the operation has any error such as syntax or semantic error
+        /// </summary>
+        public static bool IsInvalid(this IOperation operation, Compilation compilation, CancellationToken cancellationToken = default(CancellationToken))
+        {
+            // once we made sure every operation has Syntax, we will remove this condition
+            if (operation.Syntax == null)
+            {
+                return true;
+            }
+
+            // if given compilation is wrong, we will throw null ref exception
```

Comment:
I prefer to state all expected conditions explicitly in code, so that in debug mode we can immediately look at the trace and know that we are violating some precondition.